### PR TITLE
Update October-2020

### PR DESCRIPTION
**Pull request by danieleongari**
Using a new structure of the database/queries that is the same for CCS and other applications, and exploits extras on groups.
- [x] update all queries
- [x] add instructions to reproduce Outlook's figures in the `info` page
- [x] update the README with the database's API

### CODE REVIEW

**Comment by danieleongari** [comment]
@ltalirz please give a look to `figure.configure.load_profile()` if this can be moved to `pipeline_config`.
It is also weird that newer pages (details, results) are not using the `update_profile`, so I'm not sure it is still needed!

**Comment by ltalirz** [comment]
> please give a look to `figure.configure.load_profile()` if this can be moved to `pipeline_config`.

Yes, you can move it there.



> It is also weird that newer pages (details, results) are not using the `update_profile`, so I'm not sure it is still needed!

Well, this function is needed only inside the docker container and all it does is to create the profile if it is missing.
I.e. once one of the apps has created the profile, the others will be able to use it.
In general, however, it would be better if all apps were importing the `load_profile` function from the `pipeline_config`.

**Comment by danieleongari** [comment]
@ltalirz  I'll take care of it, thanks

**Comment by danieleongari** [comment]
All pages tested, now we should wait that the group extras can be exported.
https://github.com/aiidateam/aiida-core/issues/4497

**Comment by danieleongari** [comment]
One thing that is very inconvenient in the group organization is that `group.label` is like `discover_curated_cofs/{MAT_ID}` and therefore one has to always to use `.split("/")[1]`, and since the mat_id is important, this is needed many times. 
To circumvent this I created the extra `mat_id: {MAT_ID}`, but now one has to be careful that if he needs to modify the MAT_ID he modifies both. It is more convenient than using again and again `.split()` but not perfect!

**Comment by ltalirz** [comment]
> One thing that is very inconvenient in the group organization is that `group.label` is like `discover_curated_cofs/{MAT_ID}` and therefore one has to always to use `.split("/")[1]`, and since the mat_id is important, this is needed many times.

One could try to play around with the [GroupPath](https://aiida-core.readthedocs.io/en/latest/howto/data.html#organise-groups-in-hierarchies) functionality but in the end I think the right approach is what you did: to query for groups by the `mat_id` extra and to regard the group label more as a human-readable labelling on top - which should be possible to change at any time without breaking the queries.

File: figure/main.py [hunk 1]
```diff
@@ -47,7 +25,7 @@ def update_legends(p, q_list, hover):
 def get_plot(inp_x, inp_y, inp_clr):
     """Returns a Bokeh plot of the input values, and a message with the number of COFs found."""
     q_list = [config.quantities[label] for label in [inp_x, inp_y, inp_clr]]
-    results_wnone = get_data_aiida(q_list)  #returns [inp_x_value, inp_y_value, inp_clr_value, group.label]
+    results_wnone = get_data_aiida(q_list)  #returns ***
```
Comment (ltalirz) [inline figure/main.py:28]: I don't think *** is clearer here...
let's remove this comment and instead document in the docstring of `get_data_aiida` what the function returns

File: figure/main.py [hunk 2]
```diff
@@ -58,18 +36,17 @@ def get_plot(inp_x, inp_y, inp_clr):
...
     nresults = len(results)
     if not results:
-        results = [[0, 0, 0, 'no data']]
+        results = [['x', 'x', 'x', 0, 0, 0]]
         msg = "No matching COFs found."
     else:
         msg = "{} COFs found.<br> <b>Click on any point for details!</b>".format(nresults)
 
-    group_label, x, y, clrs = zip(*results)
+    mat_id, mat_name, mat_class, x, y, clrs = zip(*results)  # returned ***
```
Comment (ltalirz) [inline figure/main.py:44]: same here

File: pipeline_config/__init__.py [hunk 3]
```diff
@@ -66,44 +109,52 @@ def _clean(string):
...
-@lru_cache(maxsize=128)
+#@lru_cache(maxsize=128)
 def get_data_aiida(quantitites):
-    """Query the AiiDA database
-
-    :param quantities: tuple of quantities to project
-
-    TODO: the group version needs to be better rationalized!
-    """
+    """Query the AiiDA database for a list of quantities (other appl results)."""
```
Comment (ltalirz) [inline pipeline_config/__init__.py]: instead document here what the function returns

File: info/info.md [hunk 4]
```diff
@@ -48,3 +48,16 @@ The following force field parameters, mixed according to the [Lorentz-Bertherlot
...
 The available force fields are stored in [this file](https://github.com/lsmo-epfl/aiida-lsmo/blob/master/aiida_lsmo/calcfunctions/ff_data.yaml), together with their references, to be parsed by the Force Field Builder utility.
+
+## Reproducing previous plots
+These results are periodically updated with more COFs or some fixes (e.g., removing duplicates and correcting structures
+where we discover some wrong chemistry). To reproduce the exact same figure as in the 
+[ACS Central Science' Outlook](https://pubs.acs.org/doi/abs/10.1021/acscentsci.0c00988)
+please refer to the [version 6 of the associated Materials Cloud Archive](https://doi.org/10.24435/materialscloud:kd-wj).
+In particular you need to import the database 
+[export_discovery_cof_07Sep20.aiida](https://archive.materialscloud.org/record/file?record_id=519&file_id=46659e55-46d2-40ea-b1c9-0df0e1b676fa&filename=export_discovery_cof_07Sep20.aiida)
+and install the [version v0.3.1 of the discover-curated-cofs app](https://github.com/lsmo-epfl/discover-curated-cofs/tree/v0.3.1).
```
Comment (ltalirz) [inline info/info.md]:
```suggestion
The results are updated periodically to include new COFs or fixes (e.g., removing duplicates, correcting structures
where we discover wrong chemistry). To reproduce the exact same figure as in the 
[ACS Central Science Outlook](https://pubs.acs.org/doi/abs/10.1021/acscentsci.0c00988)
please refer to the dataset linked from the paper ([version 6 of the Materials Cloud Archive record](https://doi.org/10.24435/materialscloud:kd-wj).

In particular import the database 
[export_discovery_cof_07Sep20.aiida](https://archive.materialscloud.org/record/file?record_id=519&file_id=46659e55-46d2-40ea-b1c9-0df0e1b676fa&filename=export_discovery_cof_07Sep20.aiida)
and install the [version v0.3.1 of the discover-curated-cofs app](https://github.com/lsmo-epfl/discover-curated-cofs/tree/v0.3.1).
```

File: info/info.md [hunk 5]
```diff
@@ -48,3 +48,16 @@ The following force field parameters, mixed according to the [Lorentz-Bertherlot
...
+
+## Visit MatScreen.com
+The project continues on [www.MatScreen.com](https://matscreen.com/), including more nanoporous crystals (MOFs and zeolites), 
+and more screenings for different applications.
```
Comment (ltalirz) [inline info/info.md]:
```suggestion
and screenings for a wider range of applications.
```

File: make_export/create_groups_export.py [hunk 6]
```diff
@@ -11,64 +9,73 @@
 from aiida.orm import Group
 from aiida.tools.importexport.dbexport import export  # Updated to AiiDA v1.3.0
 
-from figure.config import load_profile
+from pipeline_config import load_profile
 load_profile()
 
 TAG_KEY = "tag4"
-CSS_GROUP_DIR = "discover_curated_cofs/"
+GROUP_DIR = "discover_curated_cofs/"
```
Comment (ltalirz) [inline make_export/create_groups_export.py]: one would typically put stuff like this into the `pipeline_config`

File: README.md [hunk 7]
```diff
@@ -1,84 +1,43 @@
...
-Use this app to generate interactive visualizations like [these](https://www.materialscloud.org/discover/cofs#mcloudHeader)
-for atomic structures and their properties.
-
-### UPDATE (September 2020)
-Now this package contains the `results` and `details` pages copied from `nanoporous_screening` to display different 
-applications for COFs. This will be used to produce a permanent reference for the ACS Central Science's Outlook. 
-
-Not to have a dependency to the `nanoporous_screening/pipeline` package, the needed functions are copied in the 
-`pipeline_emul.py` files in the `results` and `details` directories (two identical files).
+App for generating the interactive page of https://www.materialscloud.org/discover/curated-cofs.
```
Comment (ltalirz) [inline README.md]:
```suggestion
App showcasing screening of the CURATED-COFs, live at https://www.materialscloud.org/discover/curated-cofs.
```

File: README.md [hunk 8]
```diff
@@ -1,84 +1,43 @@
...
+Built from the [Structure-Property-Visualizer]https://github.com/materialscloud-org/structure-property-visualizer).
```
Comment (ltalirz) [inline README.md]:
```suggestion
Based on the [Structure-Property-Visualizer]https://github.com/materialscloud-org/structure-property-visualizer).
```

**Review by ltalirz — COMMENTED** [review]
thanks a lot @danieleongari !

just a couple of quick edits

**Pull request merged**